x-pack/plugins/ml/public/application/timeseriesexplorer/components/forecasting_modal/modal.js

```diff
@@ -23,12 +23,33 @@ import {
 } from '@elastic/eui';
 
 import { MessageCallOut } from '../../../components/message_call_out';
+import { useMlKibana } from '../../../contexts/kibana';
+import { getMlNodeCount } from '../../../ml_nodes_check/check_ml_nodes';
 import { ForecastsList } from './forecasts_list';
 import { RunControls } from './run_controls';
 
 import { FormattedMessage } from '@kbn/i18n-react';
 
 export function Modal(props) {
+  const [mlNodesAvailable, setMlNodesAvailable] = useState(false);
+  const {
+    services: {
+      mlServices: { mlApiServices },
+    },
+  } = useMlKibana();
+
+  useEffect(
+    function prepMlNodeCheck() {
+      const getNodeCount = async () => {
+        const { count } = await getMlNodeCount(mlApiServices);
+        setMlNodesAvailable(count > 0);
+      };
+
+      getNodeCount().catch(console.error);
+    },
+    [mlApiServices]
+  );
```

Comment:
```suggestion
  const mlNodesAvailable = useMemo(async () => {
    try {
      const { count, lazyNodeCount } = await getMlNodeCount(mlApiServices);
      return count !== 0 || lazyNodeCount !== 0;
    } catch (error) {
      console.error(error);
    }
  }, [mlApiServices]);
```
This could use `useMemo` instead

Reply:
Updated in 53b573ff6d8d57acfe1ca468df74f5a8d63354a3